HDDS-12884.unit tests for Ozone reconciler

babluraul opened the pull request:
-Added unit test for
ContainerChecksumTreeManager,
ReconcileContainerEventHandler


## What is the link to the Apache JIRA


https://issues.apache.org/jira/browse/HDDS-12884

jojochuang commented:
@babluraul please add jira id and update the PR description according to the template, thanks!

babluraul commented:
> @babluraul please add jira id and update the PR description according to the template, thanks!

@jojochuang I've added the JIRA ID and updated the PR description according to the template. Let me know if there's anything else needed. Thanks!

jojochuang commented on hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/checksum/TestContainerChecksumTreeManager.java:
```diff
@@ -533,4 +536,56 @@ public void testChecksumTreeFilePath() {
+    boolean changed = checksumFile.setReadable(false);
+
+    if (!changed && checksumFile.canRead()) {
+      System.out.println("Warning: Could not make file unreadable for test.");
```
Use LOG.warn() instead of System.out.println

jojochuang commented on hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/checksum/TestContainerChecksumTreeManager.java:
```diff
@@ -533,4 +536,56 @@ public void testChecksumTreeFilePath() {
+    assertTrue(ContainerChecksumTreeManager.checksumFileExist(mockContainer),
+            "Expected true since checksum file exists");
+  }
+  @Test
```
Add a space between each test cases.
```suggestion

  @Test
```

jojochuang commented on hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/checksum/TestContainerChecksumTreeManager.java:
```diff
@@ -533,4 +536,56 @@ public void testChecksumTreeFilePath() {
     assertEquals(checksumFile.getAbsolutePath(),
         ContainerChecksumTreeManager.getContainerChecksumFile(container).getAbsolutePath());
   }
+  @Test
```
```suggestion

  @Test
```

jojochuang commented on hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/checksum/TestContainerChecksumTreeManager.java:
```diff
@@ -533,4 +536,56 @@ public void testChecksumTreeFilePath() {
+
+    assertEquals(content, result.toStringUtf8(), "Checksum content mismatch");
+  }
+  @Test
```
```suggestion

  @Test
```

jojochuang reviewed (commented):
Looks good. Just a few style issues.

As Ethan mentioned in the jira, please update the jira summary to make it more descriptive.

ssulav commented on hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/checksum/TestContainerChecksumTreeManager.java:
```diff
@@ -533,4 +536,61 @@ public void testChecksumTreeFilePath() {
+
+  @Test
+  void testChecksumFileExistTrue() throws IOException {
+    Files.write(checksumFile.toPath(), "dummy content".getBytes());
```
Do not update the content of class object

ssulav commented on hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/reconciliation/TestReconcileContainerEventHandler.java:
```diff
@@ -279,6 +279,62 @@ public void testReconcileFailsWithIneligibleReplicas(State replicaState) throws
+  public void testReconcileFailsWithNotEligibleReplicas() throws Exception {
+    addContainer(RATIS_THREE_REP, LifeCycleState.CLOSED);
+
+    addReplicasToContainer(State.CLOSED, State.DELETED, State.DELETED);
```
Can we parametrize this with other states as well

babluraul replied:
The current test is already a duplicate, so adding more states likely won’t add coverage. I’ll go ahead and remove it

ssulav commented on hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/reconciliation/TestReconcileContainerEventHandler.java:
```diff
@@ -279,6 +279,62 @@ public void testReconcileFailsWithIneligibleReplicas(State replicaState) throws
+
+  @Test
+  public void testReconcileFailsDueToInvalidContainerState() throws Exception {
+    addContainer(RATIS_THREE_REP, LifeCycleState.DELETING);
```
Duplicate test

babluraul replied:
The current test is already a duplicate, so adding more states likely won’t add coverage. I’ll go ahead and remove it

ssulav commented on hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/reconciliation/TestReconcileContainerEventHandler.java:
```diff
@@ -279,6 +279,62 @@ public void testReconcileFailsWithIneligibleReplicas(State replicaState) throws
+  }
+
+  @Test
+  public void testScmNotLeader() {
```
Update the Test name

babluraul replied:
i will update

babluraul commented on hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/checksum/TestContainerChecksumTreeManager.java:
```diff
@@ -533,4 +536,61 @@ public void testChecksumTreeFilePath() {
+    Files.write(checksumFile.toPath(), "dummy content".getBytes());
+
+    Container mockContainer = mock(Container.class);
+    when(mockContainer.getContainerData()).thenReturn(container);
```
in the implemention 
 public KeyValueContainerData getContainerData()  {
    return containerData;
  }
  its returning a KeyValueContainerData type
  so in my case container which i return  is KeyValueContainer type

babluraul commented on hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/checksum/TestContainerChecksumTreeManager.java:
```diff
@@ -533,4 +536,61 @@ public void testChecksumTreeFilePath() {
+  void testChecksumFileExistFalse() {
+
+    Container mockContainer = mock(Container.class);
+    when(mockContainer.getContainerData()).thenReturn(container);
```
When the getContainerData() method of the mockContainer is called, it returns the real container object

babluraul commented:
closing since most of the changes are duplicates
